### PR TITLE
Display of a filter in operation and retaining of state of apply option on load - Filter dialog

### LINKED ISSUE #5901: Displaying a filter in operation
>Yes, this is more general than deleting a filter.  When a filter is in operation the row variable is in red, so you know a filter is being applied.  But which?  I thought - as a separate issue, perhaps youi could right-click in the rows (i.e. on the red bit) and there would be simply a label that Says Filter: Filter2 if a filter is in operation and perhaps Filter: None if no filter is currently applied?

_Originally posted by @rdstern in https://github.com/africanmathsinitiative/R-Instat/pull/5860#issuecomment-658311939_

@rdstern I suggest adding this as part of the label below the data view window which is easy to do.

### LINKED ISSUE #6180: Problem retaining the state of an apply option on reopen - filter dialog
This dialog doesn't retain the state of an apply option (Apply as Filter or Apply as Subset) selected on reopen. Our principle is that dialogs should always retain states on reopen.

### LINKED ISSUE #6181: Filter doesn't overwrite?
I have considered this as a small bug - but you may disagree?  

I had filter1 for a data frame.  It was incorrect.  So I chose to name the new filter as filter1 to replace it.  It doesn't allow me to do that.  I think it should.

We can delete filters now, so I assume this should be an easy change?

### PR DESCRIPTION
**Pull request by shadrackkibet**
Fixes #5901, Fixes #6180 and Fixes #6181. @africanmathsinitiative/developers this is ready for review. 

Notes
- Removed resetting of state for apply options on load.
- Added a method for getting the filter name.
- Setting up the display of an active filter below the current sheet.
- Clearing the filter receiver.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Ivanluv please could you review? thanks

**Comment by Ivanluv** [comment]
@lloyddewit this is fine

**Comment by shadrackkibet** [comment]
Please note that this PR now includes a fix that ensures the preview is updated correctly when the filter is overwritten.

**Comment by rdstern** [comment]
@shadrackkibet great.  I'll check

**Review by rdstern — APPROVED** [review]
@lloyddewit all fine for me.

File: instat/ucrDataView.vb [hunk 1]
```diff
@@ -343,7 +345,8 @@ Public Class ucrDataView
             iColumnCount = frmMain.clsRLink.GetDataFrameColumnCount(grdCurrSheet.Name)
             lblRowDisplay.Text = "Showing " & grdCurrSheet.RowCount & " of " & iRowCount & " rows"
             If frmMain.clsRLink.RunInternalScriptGetValue(clsFilterApplied.ToScript()).AsLogical(0) Then
-                lblRowDisplay.Text = lblRowDisplay.Text & " (" & frmMain.clsRLink.GetDataFrameLength(grdCurrSheet.Name, False) & ")"
+                Dim strFilterName As String = frmMain.clsRLink.RunInternalScriptGetValue(clsGetCurrentFilterName.ToScript(), bSilent:=True).AsCharacter(0)
+                lblRowDisplay.Text = lblRowDisplay.Text & " (" & frmMain.clsRLink.GetDataFrameLength(grdCurrSheet.Name, False) & ")" & "|Active filter:" & strFilterName
```
Comment (lloyddewit) [inline instat/ucrDataView.vb:349]: Use '&=' operator?
Could also make the same change to line 351 below
```suggestion
                lblRowDisplay.Text &= " (" & frmMain.clsRLink.GetDataFrameLength(grdCurrSheet.Name, False) & ")" & "|Active filter:" & strFilterName
```

**Review by lloyddewit — APPROVED** [review]
@shadrack this looks good, just one small comment (which I don't think isn't worth delaying the merge for)

File: instat/ucrDataView.vb (hunk 1 above)
Comment (shadrackkibet) [reply instat/ucrDataView.vb:349]: This a convenient short-hand. I like it. Thanks for the insight.

**Pull request merged**